### PR TITLE
Desk

### PR DESCRIPTION
**Pull request by NicoleZuckerman**
Contains:
-- Self check in: Includes link in profile (behind env var), uploading photo, CTA to pay balance via paypal.
-- Desk check in: Includes adding additional add-ons & accounting for balance paid through square or whatever. Also photo upload.

@NorthIsUp plz review!  I know it's not passing all the checks but I want to get us talking about what I'm doing here ASAP so I can merge sooner than later.
NOTE: I had to reconcile 2 conflicting migrations.

### CODE REVIEW

File: lindy/check_in/views.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        if success:
+            kwargs.pop('amount')
+            return HttpResponseRedirect(
+                reverse('desk_pre_check_in_purchase_view', kwargs=kwargs)
+            )
+        else:
+            return HttpResponseRedirect(
+                reverse('desk_pre_check_in_purchase_view', kwargs=self.kwargs)
+            )
+            # What should I do here?
```
Comment (NorthIsUp) [inline lindy/check_in/views.py]: elaborate?

File: lindy/check_in/views.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+        return HttpResponseRedirect(
+            reverse('pre_check_in_purchase_view', kwargs=self.kwargs)
+        )
+
+    def get_context_data(self, event_slug, ticket_id, **kwargs):
+        user_id = self.request.user.id
+        kwargs = super().get_context_data(event_slug, ticket_id, user_id, **kwargs)
+        if kwargs['ticket'].checked_in:
+            HttpResponseRedirect(
+                        reverse('confirmation', kwargs=self.kwargs)
```
Comment (NorthIsUp) [inline lindy/check_in/views.py]: weird indentation

File: lindy/check_in/views.py (hunk 1 above)
Comment (NicoleZuckerman) [inline lindy/check_in/views.py]: i.e. what should happen if we weren't able to create an IPN for some reason?

File: lindy/check_in/views.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+            self.ticket = self.get_ticket(event_slug, ticket_id, user_id)
+            if self.ticket:
+                self.ticket.attendee.photo = form.cleaned_data['photo']
+                self.ticket.attendee.save()
+        self.kwargs.update(user_id=user_id)
+
+    def get_context_data(self, event_slug, ticket_id, user_id, **kwargs):
+        self.ticket = self.get_ticket(event_slug, ticket_id, user_id)
+        if not self.ticket:
+            pass  # do some stuff here!
```
Comment (NicoleZuckerman) [inline lindy/check_in/views.py]: @NorthIsUp what do you think should happen in this case

File: lindy/check_in/views.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+    def get_context_data(self, event_slug, ticket_id, user_id, **kwargs):
+        kwargs = super().get_context_data(event_slug, ticket_id, user_id, **kwargs)
+        kwargs.update(
+            check_in_link=reverse('desk_check_in', kwargs=self.kwargs),
+            manual_payment_link = reverse('complete_manual_payment', kwargs={
+                    'event_slug': event_slug,
+                    'ticket_id': ticket_id,
+                    'user_id': user_id,
+                    'amount': str(int(self.ticket.amount_pending * 100))
```
Comment (NicoleZuckerman) [inline lindy/check_in/views.py]: Is there a different way I should be doing this-- converting the '5600' value into 56.00 decimal?

**Pull request merged**